daemon/src/main/java/bisq/daemon/grpc/interceptor/GrpcServiceRateMeteringConfig.java

```diff
@@ -0,0 +1,268 @@
+        }
+        return serviceMethodConfigurations;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static GrpcCallRateMeter getGrpcCallRateMeter(Map.Entry<String, Object> gsonEntry) {
+        Map<String, Object> valueMap = (Map<String, Object>) gsonEntry.getValue();
+        int allowedCallsPerTimeUnit = ((Number) valueMap.get(KEY_ALLOWED_CALL_PER_TIME_UNIT)).intValue();
+        TimeUnit timeUnit = TimeUnit.valueOf((String) valueMap.get(KEY_TIME_UNIT));
+        return new GrpcCallRateMeter(allowedCallsPerTimeUnit, timeUnit);
+    }
+
+    private static void verifyConfigFile(File configFile) {
+        if (configFile == null)
+            throw new IllegalStateException("Cannot read null json config file.");
+
+        if (!configFile.exists())
+            throw new IllegalStateException(format("cannot find json config file %s", configFile.getAbsolutePath()));
+    }
+
+    private static String toJson(File configFile) {
+        try {
+            return new String(readAllBytes(Paths.get(configFile.getAbsolutePath())));
+        } catch (IOException ex) {
+            throw new IllegalStateException(format("Cannot read json string from file %s.",
+                    configFile.getAbsolutePath()));
+        }
+    }
+
+    private static List<GrpcServiceRateMeteringConfig> allDeserializedConfigs;
+
+    private static List<GrpcServiceRateMeteringConfig> getAllDeserializedConfigs(File configFile) {
+        // We deserialize once, not for each gRPC service wanting an interceptor.
+        if (allDeserializedConfigs == null)
+            allDeserializedConfigs = deserialize(configFile);
+
+        return allDeserializedConfigs;
+    }
+
+    @VisibleForTesting
+    public static class Builder {
+        private final List<GrpcServiceRateMeteringConfig> rateMeterConfigs = new ArrayList<>();
+
+        public void addCallRateMeter(String grpcServiceClassName,
+                                     String methodName,
+                                     int maxCalls,
+                                     TimeUnit timeUnit) {
+            log.info("Adding call rate metering definition {}.{} ({}/{}).",
+                    grpcServiceClassName,
+                    methodName,
+                    maxCalls,
+                    StringUtils.chop(timeUnit.name().toLowerCase()));
+            rateMeterConfigs.stream().filter(c -> c.isConfigForGrpcService(grpcServiceClassName))
+                    .findFirst().ifPresentOrElse(
+                    (config) -> config.addMethodCallRateMeter(methodName, maxCalls, timeUnit),
+                    () -> rateMeterConfigs.add(new GrpcServiceRateMeteringConfig(grpcServiceClassName)
+                            .addMethodCallRateMeter(methodName, maxCalls, timeUnit)));
+        }
+
+        public File build() {
+            File tmpFile = serializeRateMeterDefinitions();
+            File configFile = new File("/tmp/ratemeters.json");
```

Comment:
Not sure this is right unless it's only supposed to work on *nix.

Reply:
It is only supposed to work on *nix but I changed line 233 to
`File configFile = Paths.get(getProperty("java.io.tmpdir"), "ratemeters.json").toFile();`